utils/counter.go

```diff
@@ -0,0 +1,43 @@
+package utils
+
+import (
+	"encoding/binary"
+
+	"github.com/axelarnetwork/axelar-core/utils/key"
+	"github.com/axelarnetwork/utils/convert"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/tendermint/tendermint/libs/log"
+	"golang.org/x/exp/constraints"
+)
+
+type Counter[T constraints.Unsigned] interface {
```

Comment:
this is not going to be passed around much, I don't think there is a lot of value in preemptively defining an interface

Reply:
k agree. I'll get rid of it, and we can bring it back when there's a real need.